### PR TITLE
Added inputs, deprecated `source_json`

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Added inputs, deprecated `source_json`
* Use latest tests

## why
* Use non-deprecated input

## references
* Closes https://github.com/cloudposse/terraform-aws-iam-policy/issues/12
* This works for aws 3.x too https://registry.terraform.io/providers/hashicorp/aws/3.75.2/docs/data-sources/iam_policy_document#override_policy_documents
* After this PR, we want to update https://github.com/cloudposse/terraform-aws-helm-release

### CODE REVIEW

File: examples/complete/variables.tf [hunk 1]
```diff
@@ -11,5 +11,5 @@ variable "iam_source_json_url" {
 variable "iam_policy_statements" {
   type        = any
   description = "List of IAM policy statements to use in the policy. This can be used with or instead of the `var.iam_source_json_url`."
```
Comment (Gowiem) [inline examples/complete/variables.tf:13]: Nitpick, but maybe we change to `Map of ...` now that the default is a map?

File: examples/complete/versions.tf [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
-    local = {
-      source  = "hashicorp/local"
-      version = ">= 1.2"
+    # Update these to reflect the actual requirements of your module
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 3.0"
+    }
+    http = {
+      source  = "hashicorp/http"
```
Comment (Gowiem) [inline examples/complete/versions.tf:11]: Good add 👍

File: main.tf [hunk 3]
```diff
@@ -14,7 +21,10 @@ data "http" "iam_source_json_url" {
 data "aws_iam_policy_document" "this" {
   count = local.enabled ? 1 : 0
 
-  source_json = var.iam_source_json_url != null ? data.http.iam_source_json_url[0].body : var.iam_source_json
+  policy_id = var.iam_policy_id
```
Comment (Gowiem) [inline main.tf:24]: Good add 👍

File: variables.tf [hunk 4]
```diff
@@ -1,20 +1,13 @@
...
 variable "iam_source_json_url" {
   type        = string
   description = "IAM source JSON policy to download and use as `source_json` argument. This is useful when using a 3rd party service that provides their own policy. This can be used with or instead of the `var.iam_policy_statements`."
   default     = null
 }
 
-# Note: cannot be a list(any) as terraform complains
 variable "iam_policy_statements" {
   type        = any
   description = "List of IAM policy statements to use in the policy. This can be used with or instead of the `var.iam_source_json_url`."
```
Comment (Gowiem) [inline variables.tf:9]: Really meant that description update comment above for here.

**Review by Gowiem — COMMENTED** [review]
@nitrocode looks like a great upgrade 👍 One nitpick on the description of the updated var, but otherwise I think it looks ready to go.

**Comment by Gowiem** [comment]
@nitrocode I imagine this should be a major version rev, correct? Do we have automation in place to make that happen or do you need to do this as a non-release and then release manually?

**Comment by nitrocode** [comment]
> @nitrocode I imagine this should be a major version rev, correct? Do we have automation in place to make that happen or do you need to do this as a non-release and then release manually?

Hmm I don't know. The patch version is for backwards compatibility. The minor version is for adding new functionality in a backwards compatible manner. And major version is for breaking changes. I suppose this is a breaking change unless I add a `variables-deprecated.tf` file and if `source_json` input is used, it uses a local to combine it with the `source_policy_documents`. I'd prefer to not release a major version for now since this module is still kind of "incubating" since it's new.

Perhaps I should go with the deprecated route to avoid breaking changes then?

**Review by Gowiem — APPROVED** [review]

**Pull request merged**